client.go

```diff
@@ -228,10 +229,13 @@ func (c *Client) fullURL(suffix string, args ...any) string {
 	if c.config.APIType == APITypeAzure || c.config.APIType == APITypeAzureAD {
 		baseURL := c.config.BaseURL
 		baseURL = strings.TrimRight(baseURL, "/")
+		parseURL, _ := url.Parse(baseURL)
```

Comment:
@eiixy btw, if you'd have any interest – it would be great to refactor this method https://github.com/sashabaranov/go-openai/issues/636